modules/lang-mustache/src/main/java/org/opensearch/script/mustache/SearchTemplateRequest.java

```diff
@@ -259,16 +259,25 @@ public void writeTo(StreamOutput out) throws IOException {
 
     @Override
     public String[] indices() {
+        if (simulate) {
```

Comment:
If the request is simulated it seems like the request is null, rather than use these guards to protect against nulls, could we have a static SIMULATED_REQUEST that has these defaults so during construction we can verify there is no request and then use the simulated version?

Reply:
Added this into `setSimulate`. In order for `simulate` to be set to true, this method must be called. The SearchTemplateRequestBuilder also uses this method when setting `simulate` to true.

Reply:
I think there is still a gap when the SearchTemplate is extracted from the StreamInput?

Reply:
I can add a guard, but I don't think that can happen since the only way that constructor is called is when a node receives a transport request from another node. As long as when the normal construction path ensures that request is set to the default value when `simulate` is set to true, then it will be serialized w/ the default request object.

I do think there should be enforcement in `setRequest` that doesn't permit null though because this code would be problematic.

```
SearchTemplateRequest r = new SearchTemplateRequest();
r.setSimulate(true);
r.setRequest(null);
```

Reply:
Added enforcement to ensure that `setRequest` cannot be called with `null`.

Reply:
I do not think adding a side effects on a setter is a good practice.  Lets come up with a way that avoids encoding business logic in the serialization object.

Reply:
Ok, I reverted this PR back to a state where there is special handling in the request object is null.

On another branch I was looking into separating these into 2 separate request objects that inherit from an AbstractSearchTemplateRequest. I'll pick that back up, but it will be a larger change then adding the handling into the existing request class and handling the scenario when simulate is set to true (same as request object being null).